client/gatsby-config.js

```diff
@@ -4,7 +4,7 @@ const {
   buildChallenges,
   replaceChallengeNode,
   localeChallengesRootDir
-} = require('./utils/buildChallenges');
+} = require('./utils/build-challenges');
```

Comment:
```suggestion
} = require('./utils/build-challenges.ts');
```
Does this not work?

Reply:
Unfortunately not.  There's a lot of discussion about this here https://github.com/Microsoft/TypeScript/issues/27481, but the TL;DR is nope: you have to omit the extension.

Anyway, the reason the existing import does not work is that we don't currently have the tooling to support this.  Everything that gets imported by the client will be transpiled by Gatsby, but if a file is used elsewhere then it won't be.

Concretely: `gatsby-config` is pure JS, so cannot import `.ts` files.

Sorry @awu43 I didn't think about this when I set up the issue.  I'll create another issue for the tooling, but for now let's hold off the migration of `buildChallenges.js`.

Reply:
Reverting `build-challenges` changes the point of failure to `./utils/gatsby`. Further reverting `gatsby/index` and `gatsby/challenge-page-creator` fixes this. Should I proceed on that?

Reply:
Yes, please.  I'll remove those files from the umbrella issue, too.